lte/gateway/c/session_manager/test/test_polling_pipelined.cpp

```diff
@@ -0,0 +1,176 @@
+    auto default_mconfig = get_default_mconfig();
+    local_enforcer       = std::make_unique<LocalEnforcer>(
+        reporter, rule_store, *session_store, pipelined_client, events_reporter,
+        spgw_client, aaa_client, 0, 0, default_mconfig);
+    evb = folly::EventBaseManager::get()->getEventBase();
+    local_enforcer->attachEventBase(evb);
+    session_map   = SessionMap{};
+    test_cfg_     = get_default_config("");
+    default_cfg_1 = get_default_config(IMSI1);
+    default_cfg_2 = get_default_config(IMSI2);
+
+    teids0.set_agw_teid(0);
+    teids0.set_enb_teid(0);
+    teids1.set_agw_teid(TEID_1_UL);
+    teids1.set_enb_teid(TEID_1_DL);
+    teids2.set_agw_teid(TEID_2_UL);
+    teids2.set_enb_teid(TEID_2_DL);
+  }
+
+  virtual void TearDown() { folly::EventBaseManager::get()->clearEventBase(); }
+
+  void run_evb() {
+    evb->runAfterDelay([this]() { local_enforcer->stop(); }, 100);
+    local_enforcer->start();
+  }
+
+  SessionConfig get_default_config(const std::string& imsi) {
+    SessionConfig cfg;
+    cfg.common_context =
+        build_common_context(imsi, IP1, IPv6_1, teids1, APN1, MSISDN, TGPP_LTE);
+    QosInformationRequest qos_info;
+    qos_info.set_apn_ambr_dl(32);
+    qos_info.set_apn_ambr_dl(64);
+    qos_info.set_br_unit(QosInformationRequest_BitrateUnitsAMBR_KBPS);
+    const auto& lte_context =
+        build_lte_context(IP2, "", "", "", "", BEARER_ID_1, &qos_info);
+    cfg.rat_specific_context.mutable_lte_context()->CopyFrom(lte_context);
+    return cfg;
+  }
+
+  void insert_static_rule(
+      uint32_t rating_group, const std::string& m_key,
+      const std::string& rule_id) {
+    rule_store->insert_rule(create_policy_rule(rule_id, m_key, rating_group));
+  }
+
+  protected:
+  std::shared_ptr<MockSessionReporter> reporter;
+  std::shared_ptr<StaticRuleStore> rule_store;
+  std::shared_ptr<SessionStore> session_store;
+  std::unique_ptr<LocalEnforcer> local_enforcer;
+  std::shared_ptr<MockPipelinedClient> pipelined_client;
+  std::shared_ptr<MockSpgwServiceClient> spgw_client;
+  std::shared_ptr<MockAAAClient> aaa_client;
+  std::shared_ptr<MockEventsReporter> events_reporter;
+  std::shared_ptr<MockPipelined> pipelined_mock;
+  SessionMap session_map;
+  SessionConfig test_cfg_;
+  SessionConfig default_cfg_1;
+  SessionConfig default_cfg_2;
+  folly::EventBase* evb;
+};
+
+TEST_F(LocalEnforcerTest, test_polling_pipelined) {
+    //insert some rules to retrieve
+    insert_static_rule(1, "", "rule1");
+    insert_static_rule(1, "", "rule2");
+    insert_static_rule(1, "", "rule3");
+    insert_static_rule(1, "", "rule4");
+    
+    CreateSessionResponse response;
+    local_enforcer->init_session(
+        session_map, IMSI1, SESSION_ID_1, get_default_config(IMSI1), response);
+    local_enforcer->update_tunnel_ids(
+        session_map,
+        create_update_tunnel_ids_request(IMSI1, BEARER_ID_1, teids1));
+    RuleRecordTable table;
+
+    auto record_list = table.mutable_records();
```

Comment:
Let's pull out `test_cfg_.common_context.ue_ipv4()` into a variable so the `create_rule_record` is a oneliner